File: src/main/java/kr/mjc/rachel/basics/jdbc/post/PostDao.java
package kr.mjc.rachel.basics.jdbc.post;

import java.util.List;

public interface PostDao {

  List<Post> listPosts(int count, int page);

  Post getPost(int postId);

  void addPost(Post post);

  int updatePost(Post post);

  int deletePost(int postId, int userId);
}

File: src/main/kotlin/kr/mjc/jacob/jdbc/post/PostDao.kt
package kr.mjc.jacob.jdbc.post

import kr.mjc.jacob.jdbc.Page

interface PostDao {
  fun list(page: Page): List<Post>

  fun getById(id: Int): Post?

  fun create(post: Post): Post

  fun update(post: Post): Int

  fun deleteById(id: Int): Int
}